### DISCUSSION TITLE
Generic variable: yawPrecision

### DISCUSSION THREAD

**Discussion by HydraheadHunter** [maruohon/minihud#264] [Ideas]
I've got a tunnel that's give or take 10,000 blocks long (I play a normal amount of minecraft), and at that scale, the yaw truncated or rounded from 0.0XXX down to 0.0 still mulitplies out and cause the player to drift. Drifting less than a block over 10,000 blocks travelled requires a yaw within 0.005 of the cardinal direction. Most people don't need this level of precision for anything so it doesn't make sense to show it to them even if they have yaw turned on, but for me this precision's important (cause I ain't got the resources to build a 10Km minecart track, and I want to be able to sprint down this hallway the entire length by just holding [W].

To this dichotomy of needs I propose a yawPrecision variable: An int which tells minihud how many decimal places to calculate and render yaw to. Default value for current behavior would be 1 (producing 0.0) wit 2 (0.00), 3 (0.000) and so on as options.

**Comment by maruohon** [comment]
This is quite likely going to become possible once the info lines get their planned rewrite.

For now, if you want you can just edit the code and build a separate mod version for yourself with the precision changed. You would just need to edit this line: https://github.com/maruohon/minihud/blob/bc9300b09070aa32334f836011159742d63e48a6/src/main/java/fi/dy/masa/minihud/event/RenderHandler.java#L698

For building the mod for MC 1.18+ you need a Java 17 JDK installed, and then running the command `./gradlew build` inside the mod repo directory should compile the mod and place it in the `build/libs/` directory.

And as for your tunnel use case, unless you have deemed Tweakeroo in its entirety too cheaty to use, the Snap Aim tweak would allow you to snap the player to the exact cardinal facing very easily.

**Reply by HydraheadHunter** [reply]
I hadn't heard of tweakeroo before you meantioned it, but looking at it, yeah, it's a bit too cheaty for me. Like if the vanilla line were real tweakeroo had stopped right at the line and then fallen forward over it. Thanks for pointing out the line I need to change, hopefully I can manage to compile this time.

**Reply by maruohon** [reply]
Everything in Tweakeroo is off by default though, so unless you are on a server that specifically prohibits having it at all, then it's just down to your self control to not enable any "too cheaty stuff" :p And it definitely has lots of such features, most players(?) probably only use a small portion of its features. I know I only use a small handful of its features at least.

**Reply by HydraheadHunter** [reply]
oh, the first thing anyone should know about me is I lack of self-control. I'd install for exactly one feature, then watch it balloon into a whole thing, yeah?

**Reply by maruohon** [reply]
Haha, okay, fair enough :)
